src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionBundleValidator.cs

```diff
@@ -68,50 +70,47 @@ private static string BuildRequestUrlForConditionalQueries(EntryComponent entry,
             return string.IsNullOrWhiteSpace(conditionalCreateQuery) ? entry.Request.Url : entry.Request.Url + "?" + conditionalCreateQuery;
         }
 
-        private async Task<string> GetResourceId(EntryComponent entry, CancellationToken cancellationToken)
+        private async Task<string> GetResourceId(EntryComponent entry, IDictionary<string, (string resourceId, string resourceType)> idDictionary, CancellationToken cancellationToken)
```

Comment:
Personal opinion: I think having this value tuple everywhere is a bit hard to read. A struct that holds ResourceType and ResourceId could be a light way to hold these values and shorten the definition a little bit?